src/views/Accessibility/AccessibiltyRequest/Documents/New/index.tsx

```diff
@@ -125,49 +159,179 @@ const New = () => {
         <div className="grid-container">
           <Formik
             initialValues={{
-              filename: '',
               file: {} as File,
-              uploadURL: ''
+              documentType: {
+                commonType: null,
+                otherType: ''
+              }
             }}
             onSubmit={onSubmit}
+            validationSchema={DocumentUploadValidationSchema}
+            validateOnBlur={false}
+            validateOnChange={false}
+            validateOnMount={false}
           >
             {(formikProps: FormikProps<FileUploadForm>) => {
+              const { errors, setFieldValue, values } = formikProps;
+              const flatErrors = flattenErrors(errors);
               return (
-                <Form onSubmit={formikProps.handleSubmit}>
-                  <h1>
-                    Upload a document to {data?.accessibilityRequest?.name}
-                  </h1>
-                  <Label htmlFor="file-upload">
-                    Choose a document to upload
-                    <FileUpload
-                      id="file-upload"
-                      name="file-upload"
-                      accept=".pdf,.txt"
-                      onChange={onChange}
-                      onBlur={() => {}}
-                      ariaDescribedBy=""
-                    />
-                  </Label>
-
-                  {selectedFile && (
-                    <div className="padding-top-2">
-                      <p>
-                        To keep CMS safe, documents are scanned for viruses
-                        after uploading. If something goes wrong, we&apos;ll let
-                        you know.
-                      </p>
-                      <Button
-                        type="submit"
-                        disabled={
-                          generateURLStatus.loading ||
-                          createDocumentStatus.loading
-                        }
-                      >
-                        Upload document
-                      </Button>
-                    </div>
+                <>
+                  {Object.keys(errors).length > 0 && (
+                    <ErrorAlert
+                      testId="document-upload-errors"
+                      classNames="margin-bottom-4 margin-top-4"
+                      heading="There is a problem"
+                    >
+                      {Object.keys(flatErrors).map(key => {
+                        return (
+                          <ErrorAlertMessage
+                            key={`Error.${key}`}
+                            errorKey={key}
+                            message={flatErrors[key]}
+                          />
+                        );
+                      })}
+                    </ErrorAlert>
                   )}
-                </Form>
+                  <PageHeading>
```

Comment:
This wont affect anything visually or functionally, but the `PageHeading` probably shouldn't be in `Formik`. A good place is probably between `.grid-container` and `Formik`

This is minor code quality

Reply:
I think it has to be, since the errors need to appear above the header (see figma)

Reply:
Ohhhh gotcha!